src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -112,6 +115,23 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     datasetToUpdate && datasetToUpdate.items.length === selectedIds.size
   )
 
+  useEffect(() => {
+    if (
+      previousDatasetToUpdate &&
+      datasetToUpdate &&
+      !isEqual(previousDatasetToUpdate, datasetToUpdate)
+    ) {
+      displayToast(SAVE_THE_DATASET_TO_CONTINUE, 'info', {
+        title: toastMessageTitle,
+        primaryButtonConfig: {
+          text: 'Save changes to Draft',
+          onClick: () => mutation.mutate(datasetToUpdate),
+        },
+      })
+    }
+    setPreviousDatasetToUpdate(datasetToUpdate)
+  }, [setDatasetToUpdate])
```

Comment:
Looks like a typo-this should be the variable itself, not the setter
```suggestion
  }, [datasetToUpdate])
```